services/src/pro/contexts/migrations/current_schema.sql

```diff
@@ -227,3 +227,25 @@ CREATE TABLE oidc_session_tokens (
     refresh_token bytea,
     refresh_token_encryption_nonce bytea
 );
+
+CREATE TYPE "InternalUploadedDatasetStatus" AS ENUM (
+    'Available',
+    'Expires',
+    'Expired',
+    'UpdateExpired',
+    'Deleted',
+    'DeletedWithError'
+);
+
+CREATE TABLE uploaded_user_datasets (
+    user_id uuid,
+    upload_id uuid,
+    dataset_id uuid,
+    status "InternalUploadedDatasetStatus" NOT NULL,
+    created timestamp with time zone NOT NULL,
+    expiration timestamp with time zone,
+    deleted timestamp with time zone,
+    delete_data boolean NOT NULL,
+    delete_record boolean NOT NULL,
+    PRIMARY KEY (user_id, dataset_id, upload_id)
```

Comment:
why is the user_id part of the primary key? every dataset should only ever point to one upload right?

Reply:
There is a table "user_uploads" with PRIMARY KEY (user_id, upload_id). So every upload has a user. This table additionally covers every upload has at most one dataset. So we could limit the primary key to that, if you like, but essentially, every upload has exactly one user and at most one dataset.

Reply:
ok so leave it in.

Another point: should the expiration be a separate table? I'd figure it to be NULL for almost every dataset because deletion is rare?